File: 백준/Bronze/17945. 통학의 신/통학의 신.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;
import java.lang.Math;
public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int a = Integer.parseInt(st.nextToken());
        int b = Integer.parseInt(st.nextToken());
        int rlt = (int)Math.sqrt(a*a-b);
        if(rlt>0)
            System.out.print((-a-rlt)+" "+(-a+rlt));
        else
            System.out.print(-a);
    }
}

File: 백준/Bronze/17945. 통학의 신/통학의 신.cc
#include <iostream>
#include <cmath>
using namespace std;
int main(void) {
    int a,b,rlt=0;
    cin>>a>>b;
    rlt=(int)sqrt(a*a-b);
    if(rlt>0)
        cout<<-a-rlt<<' '<<-a+rlt;
    else
        cout<<-a;
    return 0;
}